### PR TITLE
Initial FreeformInputWidget

### PR DESCRIPTION
**Pull request by pelusanchez**

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf-projects/cads-ui-library?pullRequest=48) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf-projects/cads-ui-library?pullRequest=48) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf-projects/cads-ui-library?pullRequest=48) it.</sub>

File: src/widgets/FreeformInputWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  required: boolean
+  details: FreeformInputWidgetDetails
+}
+
+interface FreeformInputWidgetProps {
+  configuration: FreeformInputWidgetConfiguration
+  /**
+   * Permitted selections for the widget.
+   */
+  constraints?: string[]
```
Comment (valentinogagliardi) [inline src/widgets/FreeformInputWidget.tsx:59]: I think we can get rid of this for now if constraints are not going to be already applied to this widget.

File: cypress/component/FreeformInputWidget.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,366 @@
+import React from 'react'
+
+import { FreeformInputWidget, TooltipProvider } from '../../src'
```
Comment (valentinogagliardi) [inline cypress/component/FreeformInputWidget.cy.tsx:3]: I think we need to make sure this widget works also when it's inside an `ExclusiveGroupWidget`.  See https://github.com/ecmwf-projects/cads-ui-library/blob/main/cypress/component/ExclusiveGroupWidget.cy.tsx for examples.

Comment (valentinogagliardi) [reply cypress/component/FreeformInputWidget.cy.tsx:3]: Also, https://github.com/ecmwf-projects/cads-ui-library/blob/main/src/utils/widgetFactory.tsx

Comment (pelusanchez) [reply cypress/component/FreeformInputWidget.cy.tsx:3]: Added at https://github.com/ecmwf-projects/cads-ui-library/pull/48/commits/667e176e034e7babb0e181a3a60af87ac4a16252 .

File: src/widgets/FreeformInputWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+          {label}
+        </WidgetTitle>
+        <WidgetTooltip
+          helpText={help || null}
+          triggerAriaLabel={`Get help about ${label}`}
+        />
+      </WidgetHeader>
+      <ReservedSpace data-stylizable='widget freeform-input reserved-error-space'>
+        {required && value?.length ? (
+          <Error>The field is required.</Error>
```
Comment (valentinogagliardi) [inline src/widgets/FreeformInputWidget.tsx:147]: I think it should be "This field is required"

Comment (valentinogagliardi) [reply src/widgets/FreeformInputWidget.tsx:147]: Also, it seems that the message does not disappear even if the value is present.

Comment (pelusanchez) [reply src/widgets/FreeformInputWidget.tsx]: Fixed!

File: CHANGELOG.md
Comment (valentinogagliardi) [inline CHANGELOG.md:1]: Please, update the `CHANGELOG` as well.

**Review by valentinogagliardi — APPROVED** [review]

**Comment by valentinogagliardi** [comment]
The initial version of `FreeformInputWidget` without validation looks good to me. You can release the major version.

**Comment by pelusanchez** [comment]
@valentinogagliardi Done, created v7.0.0 and added to cads-webportal: https://github.com/ecmwf-projects/cads-webportal/pull/148/commits/d52be1d42681b948655ecae9ca7084f8bdd4bdad

**Pull request merged**